From 71a55fb0f72c5bba26ac77f841a716d00f149c28 Mon Sep 17 00:00:00 2001
From: Miles Granger <miles59923@gmail.com>
Date: Wed, 17 Aug 2022 13:15:45 +0200
Subject: [PATCH] Exclude Python conftest and tests when PYARROW_INSTALL_TESTS
 not on

---
 python/setup.py | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/python/setup.py b/python/setup.py
index 97e0f07720bb..1d03a47ee42b 100755
--- a/python/setup.py
+++ b/python/setup.py
@@ -623,12 +623,14 @@ def has_ext_modules(foo):
 if strtobool(os.environ.get('PYARROW_INSTALL_TESTS', '1')):
     packages = find_namespace_packages(include=['pyarrow*'])
 else:
-    packages = find_namespace_packages(include=['pyarrow*'],
-                                       exclude=["pyarrow.tests*"])
+    packages = find_namespace_packages(
+        include=["pyarrow*"],
+        exclude=["pyarrow.tests*", "pyarrow.conftest"])
 
 
 setup(
     name='pyarrow',
+    package_dir={"": "."},
     packages=packages,
     zip_safe=False,
     package_data={'pyarrow': ['*.pxd', '*.pyx', 'includes/*.pxd']},
